### PR TITLE
CSS-4606 Make TLS a requirement for Trino Charm

### PR DESCRIPTION
**Pull request by AmberCharitos**
- Removal of TLS if statements in the `config.jinja` and `charm.py` 
- Add a `ready_to_start()` method which identifies before starting trino if a certificate relation is present
- General clean up around TLS parameter validation

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -111,6 +111,22 @@ def _restart_trino(self, container):
         container.restart(self.name)
         self.unit.status = ActiveStatus()
 
+    def ready_to_start(self):
+        """Check if TLS is enabled and peer relations established
+        
+        Returns:
+            True if TLS enabled and peer relation established, else False.
+        """
+        if not self._state:
```
Comment (kelkawi-a) [inline src/charm.py:120]: You can use the method ```is_ready()``` from the state class to make it ```self._state.is_ready()```.

File: src/charm.py [hunk 2]
```diff
@@ -332,53 +310,86 @@ def _push_file(self, container, context, jinja_file, path, permission=0o644):
...
         container = self.unit.get_container(self.name)
         if not container.can_connect():
             event.defer()
             return
 
-        logger.info("configuring trino")
-        log_options = {"log-level": "LOG_LEVEL"}
-        log_context = {config_key: self.config[key] for key, config_key in log_options.items()}
-        _ = self._push_file(container, log_context, "logging.jinja", "/etc/trino/log.properties")
+        if not self.ready_to_start():
```
Comment (kelkawi-a) [inline src/charm.py:374]: Perhaps the contents of this method can be moved to a general ```_validate()``` method which checks relations and config parameters? This way, all validations can clearly be observed in one place. We do this [here](https://github.com/canonical/temporal-k8s-operator/blob/dd6ad30d0af6c715f4e5f30c31bc1c0dfdecc628/src/charm.py#L208) in the Temporal charm. Once validation is done, you can proceed with starting your workload container.

File: src/literals.py [hunk 3]
```diff
@@ -12,3 +12,5 @@
 CATALOG_PATH = "/etc/trino/catalog"
 CONFIG_JINJA = "config.jinja"
 CONFIG_PATH = "/etc/trino/config.properties"
+LOG_PATH = "/etc/trino/log.properties"
+LOG_JINJA = "logging.jinja"
```
Comment (kelkawi-a) [inline src/literals.py:16]: Missing new line.

**Review by kelkawi-a — APPROVED** [review]
LGTM with some minor comments. Might be worth starting to refactor the code slightly as the charm's functionality starts becoming more complex. Nice work!

File: src/literals.py (hunk 3 above)
Comment (AmberCharitos) [reply src/literals.py]: Thanks, added.

File: src/charm.py (hunk 2 above)
Comment (AmberCharitos) [reply src/charm.py:374]: I take your point. I need to do some refactoring as the charm is growing in complexity. I'll keep this in mind when doing this refactoring PR.

File: src/charm.py (hunk 1 above)
Comment (AmberCharitos) [reply src/charm.py]: Updated, thanks.

**Review by gtato — APPROVED** [review]

**Pull request merged**